autoload/arduino.vim

```diff
@@ -648,6 +650,12 @@ function! arduino#GetPorts() abort
       call add(ports, port)
     endfor
   endfor
+  if s:OS ==? 'Windows'
+    let boards_data = json_decode(system('arduino-cli board list --format json'))
+    for board in boards_data
+      call add(ports, board['port']['address'])
+    endfor
+  endif 
```

Comment:
Does this have to be windows-only? It seems like this API should work on all platforms. We should put it behind a `if g:arduino_use_cli` check, though. And also deduplicate any entries in the list.